qtdialogs/DlgSetupManager.py

```diff
@@ -0,0 +1,1303 @@
+#  See LICENSE-MIT or https://opensource.org/licenses/MIT                    #
+#                                                                            #
+##############################################################################
+
+import os
+from io import BytesIO
+from struct import pack
+from types import MethodType
+from qtpy import QtCore, QtWidgets, QtGui
+from armoryengine.ArmoryUtils import BTC_HOME_DIR, ARMORY_DB_DIR, \
+   ARMORY_HOME_DIR, CLI_OPTIONS, LOGEXCEPT, LOGINFO, RightNow
+from armoryengine.Settings import TheSettings
+from armoryengine.CppBridge import TheBridge
+from armoryengine.BDM import TheBDM
+from armoryengine.WalletUtils import WalletMap
+from ui.QtExecuteSignal import TheSignalExecution
+
+from qtdialogs.ArmoryDialog import ArmoryDialog
+from qtdialogs.DlgWalletMigration import DlgWalletMigration
+from qtdialogs.qtdefines import QRichLabel, applyDialogBaseStyle, \
+   makeCenteredCell, makeCheckboxCell, makeButtonCell, addPlaceholderRow, \
+   selectDirectoryForQLineEdit
+
+# --- Dialog-specific constants ---
+MINIMUM_DIALOG_WIDTH = 300
+MINIMUM_DIALOG_HEIGHT = 500
+
+# Wallet state enums (from bridge)
+WALLET_STATE_UNKNOWN = 0
+WALLET_STATE_LEGACY = 1
+WALLET_STATE_MIGRATED = 2
+WALLET_STATE_ENCRYPTED = 3
+WALLET_STATE_LOADED = 5
+
+###############################################################################
+class DlgSetupManager(ArmoryDialog):
+   def __init__(self, parent=None, main=None, testing=False):
+      """Initialize the setup manager dialog."""
+      super().__init__(parent)
+
+      self.initMemberVariables(main)
+      self.setupDialogProperties()
+      self.initTabs()
+      self.setupMainLayout()
+      self.loadSettings(testing)
+      self.connectSignals()
+      # Initial populate will be done by onBridgeReady callback
+      # Bridge readiness will be handled by callback from main script
+
+   def initMemberVariables(self, main):
+      """Initialize all member variables with default values."""
+      self.main = main
+      # Visual custom styles removed; use standard Qt widgets
+      # Core lifecycle is handled by the bridge; no local process object
+
+      # UI component references
+      self.tabWidget = None
+      self.walletTab = None
+      self.coreTab = None
+      self.databaseTab = None
+      self.acceptButton = None
+      self.cancelButton = None
+
+      # Callback for spawning main window
+      self.mainWindowSpawner = None
+
+      # Create wallet manager for migration functionality
+      # This allows DlgWalletMigration to work without requiring
+      # ArmoryMainWindow
+      self.walletManager = WalletMap(self)
+
+      # Alias for compatibility with DlgWalletMigration expectations
+      self.wallets = self.walletManager
+
+      # Entropy accumulator for DlgWalletMigration compatibility
+      # This is used by registerWidgetActivateTime for cryptographic randomness
+      self.entropyAccum = BytesIO()
```

Comment:
No. extra entropy is only ever used when creating a new wallet. Wallet migration means reading an existing seed from a legacy wallet. Remove all reference to this object and all associated methods from setup manager.

As a general rule, do not touch anything that remotely has to do with cryptography without first explicitly consulting with me.